src/interfaces/token.ts

```diff
@@ -191,157 +187,76 @@ export class TokenInterface<C extends ChainId> extends ServiceInterface<C> {
+   * @param spenderAddress
    * @param amount
-   * @param account
-   * @returns transaction
+   * @param overrides
+   * @returns TokenAllowance
    */
-  async approveDeposit(
-    address: Address,
-    vaultToken: Address,
-    token: Address,
+  async approve(
+    ownerAddress: Address,
+    tokenAddress: Address,
+    spenderAddress: Address,
     amount: Integer,
-    account: Address
-  ): Promise<TransactionResponse | boolean> {
-    // If Ether is being sent, no need for approval
-    if (EthAddress === token) return true;
-
-    const signer = this.ctx.provider.write.getSigner(account);
-
-    if (vaultToken === token) {
-      const tokenContract = new Contract(token, TokenAbi, signer);
-      const partnerAddress = await this.yearn.services.partner?.address;
-      const addressToApprove = (this.shouldUsePartnerService(address) && partnerAddress) || address;
-      const tx = await tokenContract.populateTransaction.approve(addressToApprove, amount);
-      return this.yearn.services.transaction.sendTransaction(tx);
-    }
+    overrides?: CallOverrides
+  ): Promise<TransactionResponse> {
+    if (isNativeToken(tokenAddress)) throw new SdkError(`Native tokens cant be approved: ${tokenAddress}`);
+    if (tokenAddress === spenderAddress) throw new SdkError(`Cant approve token as its spender: ${tokenAddress}`);
 
-    const gasPrice = await this.yearn.services.zapper.gas();
+    const signer = this.ctx.provider.write.getSigner(ownerAddress);
+    const tokenContract = new Contract(tokenAddress, TokenAbi, signer);
+    const tx = await tokenContract.populateTransaction.approve(spenderAddress, amount, overrides);
 
-    const gasPriceFastGwei = new BigNumber(gasPrice.fast).times(new BigNumber(10 ** 9));
+    return this.yearn.services.transaction.sendTransaction(tx);
+  }
 
-    const zapProtocol = PickleJars.includes(address) ? ZapProtocol.PICKLE : ZapProtocol.YEARN;
+  /**
+   * Fetches supported zapper tokens and sets their icon
+   * @returns zapper tokens with icons
+   */
+  private async getZapperTokensWithIcons(): Promise<Token[]> {
+    const zapperTokens = await this.yearn.services.zapper.supportedTokens();
 
-    const zapInApprovalState = await this.yearn.services.zapper.zapInApprovalState(account, token, zapProtocol);
+    const zapperTokensAddresses = zapperTokens.map(({ address }) => address);
 
-    if (!zapInApprovalState.isApproved) {
-      const zapInApprovalParams = await this.yearn.services.zapper.zapInApprovalTransaction(
-        account,
-        token,
-        gasPriceFastGwei.toString(),
-        zapProtocol
-      );
-      const transaction: TransactionRequest = {
-        to: zapInApprovalParams.to,
-        from: zapInApprovalParams.from,
-        gasPrice: zapInApprovalParams.gasPrice,
-        data: zapInApprovalParams.data as string
-      };
-      return signer.sendTransaction(transaction);
-    }
+    const zapperTokensIcons = await this.yearn.services.asset.ready.then(() =>
```

Comment:
why are we using a promise here?

Reply:
Not sure why git is displaying this part of the code as changes, since I did not change `getZapperTokensWithIcons()` function at all. Maybe some parts of the code removed where duplicated and now detects them like if its changed here.